web4/treasury-web4/src/lib.rs

```diff
@@ -31,8 +32,21 @@ impl Contract {
             .then(
                 Self::ext(env::current_account_id())
                     .with_attached_deposit(env::attached_deposit())
-                    .update_widgets_callback(widget_reference_account_id, social_db_account_id),
-            )
+                    .update_widgets_callback(
+                        widget_reference_account_id,
+                        social_db_account_id.clone(),
+                    ),
+            );
+
+        if set_social_metadata_defaults.unwrap_or(false) {
+            promise = promise.then(self.internal_set_social_metadata(
```

Comment:
I would expect to see `set_social_metadata` here. If `internal_set_social_metadata` works, then it is bad and means that anyone can call it.

Reply:
`internal_set_social_metadata` does not have the `pub` keyword, so it is not exposed in the ABI. 

When the factory calls `update_widgets`, then the predecessor will not be the web4 contract itself, and then the `internal_set_social_metadata` is needed.

I see though that `update_widgets` can be called by anyone, so it should be limited to the factory and the web4 contract itself. I will change that.

Reply:
Sorry, I misread the code and I thought that you were making a new promise to call `internal_set_social_metadata`, but in fact you are calling `internal_set_social_metadata` inline, so it returns the promise that is then chained.

Reply:
Still it is good that you made me aware that `update_widgets` could be called by anyone. It now has a restriction so that it can only be called by `treasury-factory.near` and its own account.